dask/dataframe/core.py

```diff
@@ -543,6 +548,70 @@ def quantiles(self, *args, **kwargs):
                                   "Also, quantiles will now be specified on the range "
                                   "[0, 1], not [0, 100]")
 
+    def resample(self, rule, how='mean', axis=0, fill_method=None, closed=None,
+                 label=None, convention='start', kind=None, loffset=None,
+                 limit=None, base=0):
+        """Group by a DatetimeIndex values in time periods of size `rule`.
+
+        Parameters
+        ----------
+        rule : str or pandas.datetools.Tick
+            The frequency to resample by. For example, 'H' is one hour
+            intervals.
+        how : str or callable
+            Method to use to summarize your data. For example, 'mean' takes the
+            average value of the Series in the time interval `rule`.
+
+        Notes
+        -----
+        For additional argument descriptions please consult the pandas
+        documentation.
+
+        Returns
+        -------
+        dask.dataframe.Series
+
+        See Also
+        --------
+        pandas.Series.resample
+        """
+        # TODO: Probably a better way to do this
+        start, end = pd.tseries.resample._get_range_edges(self.divisions[0],
+                self.divisions[-1], rule)
+        index = pd.date_range(start=start, end=end, freq=rule)
+        newdivs = [start]
+        rule = pd.datetools.to_offset(rule)
+        for div in self.divisions[1:-1]:
+            # TODO: can we compute this value in O(1) instead of O(log n)?
+            pos = np.searchsorted(index.values, div.asm8, side='right')
```

Comment:
``rule.rollforward`` may be used to get new divisions. Though offset is not upldated.

```
# Assuming index[3] is division value
s = pd.Series([1, 2, 3, 4, 5], index=pd.date_range('2011-01-01', freq='D', periods=5))

rule = pd.datetools.to_offset('A')
rule.rollforward(s.index[3])                                          
# Timestamp('2011-12-31 00:00:00')

rule = pd.datetools.to_offset('H')
rule.rollforward(s.index[3])
# Timestamp('2011-01-04 00:00:00', offset='D')
```